sql/updates/latest-dev.sql

```diff
@@ -162,3 +162,25 @@ DROP FUNCTION IF EXISTS _timescaledb_internal.get_chunk_relstats;
 DROP FUNCTION IF EXISTS _timescaledb_internal.get_chunk_colstats;
 
 
+DROP FUNCTION IF EXISTS @extschema@.add_dimension(
+    REGCLASS,
+    NAME,
+    INTEGER,
+    ANYELEMENT,
+    REGPROC,
+    BOOLEAN
+);
+
+CREATE OR REPLACE FUNCTION @extschema@.add_dimension(
+    hypertable              REGCLASS,
+    column_name             NAME,
+    number_partitions       INTEGER = NULL,
+    chunk_time_interval     ANYELEMENT = NULL::BIGINT,
+    partitioning_func       REGPROC = NULL,
+    if_not_exists           BOOLEAN = FALSE,
+    correlated              BOOLEAN = FALSE
+) RETURNS TABLE(dimension_id INT, schema_name NAME, table_name NAME, column_name NAME, created BOOL)
+AS '@MODULE_PATHNAME@', 'ts_dimension_add' LANGUAGE C VOLATILE;
+
+ALTER TABLE _timescaledb_catalog.dimension ADD COLUMN correlated BOOLEAN;
+UPDATE _timescaledb_catalog.dimension SET correlated = FALSE;
```

Comment:
In update/downgrade SQL scripts when changing metadata tables we need to rebuild it to prevent problems with attnums. Have a look in this last PR that change metadata: https://github.com/timescale/timescaledb/pull/6624

Reply:
yeah, I had left the tedious part for the end :-D